index.html

```diff
@@ -394,11 +396,23 @@ <h4>Computation steps</h4>
                     <li>For <code>:after</code> pseudo elements, <a class="termref">User agents</a> MUST append <abbr title="Cascading Style Sheets">CSS</abbr>  textual content, without a space, to the textual content of the <code>current node</code>. </li>
                   </ul>
                 </li>
-                <li id="step2F.iii">For each child node of the <code>current node</code>:
+                <li id="step2F.iii">
+                  Determine the <code>rendered child nodes</code> of the <code>current node</code>:
                   <ol>
-                    <li id="step2F.iii.a">Set the <code>current node</code> to the child node.</li>
-                    <li id="step2F.iii.b">Compute the text alternative of the  <code>current node</code> beginning with step 2.  Set the <code>result</code> to that text alternative.</li>
-                    <li id="step2F.iii.c">Append the <code>result</code> to the <code>accumulated text</code>. </li>
+                    <li id="step2F.iii.a">If the <code>current node</code> has an attached [=shadow root=], set the <code>rendered child nodes</code> to be the child nodes of the [=shadow root=].</li>
+                    <li id="step2F.iii.b">Otherwise, if the <code>current node</code> is a [=slot=] with [=slot/assigned nodes=], set the <code>rendered child nodes</code> to be the [=slot/assigned nodes=] of the <code>current node</code>.</li>
```

Comment:
My intention is to tweak the children-traversal part of the algorithm to traverse into slot assigned nodes rather than child nodes. The slot itself is not excluded from processing; you're correct.

Based on https://github.com/w3c/html-aam/issues/440 it looks like this should be handled in the `html-aam` spec instead?

Reply:
> The slot itself is not excluded from processing; you're correct.

If I understand correctly, that means the Chromium bug isn't dealt with by this addition to the spec, correct? I'm not saying this isn't necessary, just clarifying for sure that it doesn't address the Chromium bug.

> it looks like this should be handled in the html-aam spec instead?

This gets back to my comment about tree traversal. As I see it, it's less to do with the fact that a slot can't be named and more to do with the fact that the slot itself (but not its rendered children) should be skipped in the traversal in the first place. I realise this might sound like nitpicking, but I think it's important in understanding how browsers actually do this.

Reply:
Oh blah. You're right: it's probably the fact that it has display: contents that excludes it from the Firefox a11y tree. As your test case shows, it does get exposed with display: block (because block elements have some semantic value by virtue of being block). That means that `<slot aria-label="foo">` probably would get exposed in Firefox except for [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1494196).

So it does look like we'll need some explicit provision that slots shouldn't be exposed. I'm not sure if them not having a name is enough. That said, this is a bit tricky because `<slot style="display: block;" tabindex="0">` does become focusable.

Reply:
Thanks, that makes sense! To identify this bug, we could try adding WPT tests that set `display: block` on the `<slot>`.

> That said, this is a bit tricky because `<slot style="display: block;" tabindex="0">` does become focusable.

Well, maybe `<slot>` _should_ be an element that can support naming? Unless https://github.com/w3c/accname/issues/173#issuecomment-1279626528 is the last word on this.

Note that I don't have a strong opinion; I'm just trying to determine the correct behavior.

Reply:
In the case someone does `<slot style="display: block;" tabindex="0">`, we've gotten quite into the territory of author error correction.

If properly used, the slot element, along with elements like title, meta, template, etc... none of those should ever actually be exposed to a user and they should not be nameable.  But, if you force these elements to be rendered by changing their CSS display property, then they essentially become 'generic' elements.

I think it makes sense to call out that if an author forces elements like these to be exposed to the a11y tree, then browsers should handle them like generics.  But, again, I would stress this is correcting for author misuse and otherwise - if used properly - they cannot be named.

Reply:
> I think it makes sense to call out that if an author forces elements like these to be exposed to the a11y tree, then browsers should handle them like generics.

@scottaohara Thanks for the feedback! Do you think this should be done in `accname` or `html-aam`? I'm wondering if it makes more sense to put it in `html-aam` (per https://github.com/w3c/html-aam/issues/440), since in `accname` we're merely defining the tree traversal algorithm w.r.t `<slot>`s and their assigned nodes, not whether `<slot>`s can be named.

Reply:
yes, i think this should be handled in html aam per the linked issue.  i don't think accName should need to call this particular element out in any specific way. that's is what html aam can do.